pkg/api/converterfromapi.go

```diff
@@ -169,10 +170,30 @@ func convertPropertiesToV20160930(api *Properties, p *v20160930.Properties) {
 		convertMasterProfileToV20160930(api.MasterProfile, p.MasterProfile)
 	}
 	p.AgentPoolProfiles = []*v20160930.AgentPoolProfile{}
-	for _, apiProfile := range api.AgentPoolProfiles {
-		v20160930Profile := &v20160930.AgentPoolProfile{}
-		convertAgentPoolProfileToV20160930(apiProfile, v20160930Profile)
-		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20160930Profile)
+	// DCOS conversion logic
+	if api.OrchestratorProfile.IsDCOS() && len(api.AgentPoolProfiles) == 2 {
```

Comment:
This condition is meant to be an exact inverse pair with this:

https://github.com/Azure/acs-engine/pull/787/files#diff-fab6595d1729597cff2aeb9912285552R206